components/emoji_picker/emoji_picker_overlay.jsx

```diff
@@ -73,11 +55,23 @@ export default class EmojiPickerOverlay extends React.PureComponent {
         return rightOffset;
     }
 
+    getPlacement() {
+        const target = this.props.target();
+        if (target) {
+            const targetBounds = target.getBoundingClientRect();
+            return popOverOverlayPosition(targetBounds, window.innerHeight, this.props.spaceRequiredAbove, this.props.spaceRequiredBelow);
+        }
+
+        return 'top';
+    }
+
     render() {
+        const placement = this.getPlacement();
+        const rightOffset = this.emojiPickerPosition();
         return (
             <Overlay
                 show={this.props.show}
-                placement={this.state.placement}
+                placement={placement}
```

Comment:
asking for `getBoundingClientRect()` in render can cause performance issues because of JS reflow . Can we add this back to be based on state?

Reply:
What issues are you worried about? The function is only called once in `render`, and that shouldn't trigger the `EmojiPickerOverlay` itself to re-render, so I'm not sure where those issues would be

Reply:
I meant that as a general rule of thumb. If someone adds a new prop which can cause extra render cycles on typing etc then we will hit the issue.

But apparently we do call the render of this a lot more than you think somehow. I just put up a break point in the render of the statement on community and it gets called a lot even when there is no emoji picker. Hover on a post causes this to be called along with log of socket events.

![Oct-23-2019 20-01-50](https://user-images.githubusercontent.com/4973621/67403801-1670ba80-f5d0-11e9-923d-c58ca6652c6d.gif)

Reply:
Have put placement and rightOffset back into state and set them in getDerivedStateFromProps